libs/dal/src/repositories/tenant/tenant.entity.ts

```diff
@@ -0,0 +1,22 @@
+import { TenantCustomData } from '@novu/shared';
+import { TenantId } from './types';
+import { EnvironmentId } from '../environment';
+import { ChangePropsValueType } from '../../types/helpers';
+
+export class TenantEntity {
+  _id: TenantId;
+
+  identifier: string;
+
+  name?: string;
+
+  deleted?: boolean;
+
+  createdAt: string;
+
+  data?: TenantCustomData;
```

Comment:
Additional custom data that the user can add.

Reply:
Wondering if we should define the expected structure even if we have not a clear idea of what to allow. For example right now with the custom data type we are allowing array of strings but not array of numbers. And no nested objects.
Another point is if the needs for the subscriber custom data might be different for the tenant. I guess something to discuss soon in the future.

Reply:
I believe `unknown` could do the job, but IMO it depends on how we will use it internally, if it will be used in the template-editor/integration-condition/template-filter, etc because if we, for example, won't support their nested object but will support in this object it could confuse the user. in one place he can store data but he won't be able to use it across the app.

meanwhile, i added the TenantCustomData in order to maintain/reuse the current BL :)